### ISSUE TITLE
Engineering work for vscode-test and @types/vscode

### ISSUE THREAD

**Issue by octref** [microsoft/vscode#71048]
In continuation of #70175. There a few things I'd want to sort out before we can recommend people switching to vscode-test and `@types/vscode`

Already done:

- Published `@types/vscode`: 1.14 - 1.32
- https://github.com/microsoft/vscode-test
- https://github.com/octref/vscode-test-sample, with Travis and DevOps configurations

For `@types/vscode`

- [x] Write an Azure function that automates the publishing of `@types/vscode` when pushing tag to vscode release branch.
- [x] Update samples https://github.com/Microsoft/vscode-extension-samples
- [x] Provide a module `vscode-dts`, so we can run `npx vscode-dts dev` or `npx vscode-dts 1.30` etc for testing: https://github.com/microsoft/vscode-dts
- [x] vsce for validating @types/vscode versions against engines.vscode: https://github.com/microsoft/vscode-vsce/pull/354

For vscode-test:

- [ ] ~~Remove lib/shared and update install script https://github.com/Microsoft/vscode-extension-vscode/issues/154~~
- [x] Move test runner to `vscode-test` https://github.com/Microsoft/vscode-extension-vscode/issues/157
- [x] Update https://code.visualstudio.com/api/working-with-extensions/testing-extension and https://code.visualstudio.com/api/working-with-extensions/continuous-integration: https://github.com/microsoft/vscode-docs/pull/2695
- [x] Upgrade generator

**Comment by Xanewok** [comment]
Thanks a lot @octref for the related work on supporting the vscode types and test runner in a more npm-first-class manner! This seemed like a wart last time I refreshed the [rls-vscode](https://github.com/rust-lang/rls-vscode) extension (Rust support), so I'd consider the work very needed.

Is there a timeline on when it is recommended to switch over to `@types/vscode` and `vscode-test`?

**Comment by octref** [comment]
@Xanewok You can wait until next Wednesday for new release, or read the to-be-released docs here:
- https://github.com/microsoft/vscode-docs/blob/vnext/release-notes/v1_36.md#splitting-vscode-package-into-typesvscode-and-vscode-test
- https://github.com/microsoft/vscode-docs/blob/vnext/api/working-with-extensions/testing-extension.md

> Maybe it's safe already to switch to @types/vscode already?

Yes, you can switch already.
